actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
`ActiveStorage::Variant` doesn't respond to `to_model`, it doesn't have to: https://github.com/rails/rails/blob/d3c0dcff20f93889fedf1ff6a5be4ea8370132fe/actionpack/lib/action_dispatch/routing/polymorphic_routes.rb#L169-L175